pkg/register/restResolver.go

```diff
@@ -23,20 +22,41 @@ import (
 
 var defaultResolverTimeout = time.Second * 60
 
+// ResolverClientMetrics interface
+type ResolverClientMetrics interface {
+	// RecordError records errors occurred during resolver calls
```

Comment:
> records errors **_occurred_**

.. doesn't sound quite right to me..

encountered / which have occurred / that occurred